tests/DbTestCase.php

```diff
@@ -51,10 +53,29 @@ public function setUp() {
       ];
    }
 
+   public function beforeTestMethod($method) {
+      if (in_array($method, $this->cached_methods)) {
+         $this->nscache = 'glpitestcache' . GLPI_VERSION;
+         global $GLPI_CACHE;
+         //run with cache
+         define('CACHED_TESTS', true);
+         $GLPI_CACHE = \Zend\Cache\StorageFactory::factory([
+            'adapter'   => 'apcu',
+            'options'   => [
+               'namespace' => $this->nscache
+            ]
+         ]);
+      }
+   }
 
    public function afterTestMethod($method) {
       global $DB;
 
+      if (in_array($method, $this->cached_methods)) {
+         global $GLPI_CACHE;
+         $GLPI_CACHE = false;
+      }
```

Comment:
Shouldn't you also call apcu_clear_cache, to ensure store values from one cache will not polute another TU ?

Reply:
rather $GLPI_CACHE::flush();